src/storage/redis_db.cc

```diff
@@ -96,7 +96,21 @@ rocksdb::Status Database::GetMetadata(engine::Context &ctx, RedisTypes types, co
   auto s = GetRawMetadata(ctx, ns_key, raw_value);
   *rest = *raw_value;
   if (!s.ok()) return s;
-  return ParseMetadataWithStats(types, rest, metadata);
+
+  s = ParseMetadataWithStats(types, rest, metadata);
+  if (!s.ok()) return s;
+
+  // if type is hash, we still need to check if the all of fields expired.
+  if (metadata->Type() == kRedisHash) {
+    HashMetadata hash_metadata(false);
+    s = hash_metadata.Decode(*raw_value);
+    if (!s.ok()) return s;
+    redis::Hash hash_db(storage_, namespace_);
+    if (!hash_db.ExistValidField(ctx, ns_key, hash_metadata)) {
+      return rocksdb::Status::NotFound("no element found");
+    }
+  }
```

Comment:
This will make almost all operation of HASH inefficient, we should consider how to do this in another way.

Reply:
It requires at least O(N) IO accesses for all HASH operations. And if you think about it: not all operations require to check that all elements are expired before the actual operation.